src/colorize.cr

```diff
@@ -226,108 +231,98 @@ struct Colorize::Object(T)
-  private MODE_UNDERLINE_FLAG =  4
-  private MODE_BLINK_FLAG     =  8
-  private MODE_REVERSE_FLAG   = 16
-  private MODE_HIDDEN_FLAG    = 32
-
   private COLORS = %w(default black red green yellow blue magenta cyan light_gray dark_gray light_red light_green light_yellow light_blue light_magenta light_cyan white)
   private MODES  = %w(bold bright dim underline blink reverse hidden)
 
-  @fore : Color
-  @back : Color
-
   def initialize(@object : T)
     @fore = ColorANSI::Default
     @back = ColorANSI::Default
-    @mode = 0
+    @mode = Mode::None
     @enabled = Colorize.enabled?
   end
 
-  {% for name in COLORS %}
-    def {{name.id}}
-      @fore = ColorANSI::{{name.camelcase.id}}
+  {% for color in COLORS %}
+    def {{color.id}}
+      @fore = ColorANSI::{{color.camelcase.id}}
       self
     end
 
-    def on_{{name.id}}
-      @back = ColorANSI::{{name.camelcase.id}}
+    def on_{{color.id}}
+      @back = ColorANSI::{{color.camelcase.id}}
       self
     end
   {% end %}
 
-  {% for name in MODES %}
-    def {{name.id}}
-      @mode |= MODE_{{name.upcase.id}}_FLAG
+  def on(color : Color)
+    back color
+  end
+
+  {% for mode in MODES %}
+    def {{mode.id}}
+      @mode |= Mode::{{mode.capitalize.id}}
       self
     end
   {% end %}
 
-  def fore(color : Symbol)
-    {% for name in COLORS %}
-      if color == :{{name.id}}
-        @fore = ColorANSI::{{name.camelcase.id}}
-        return self
-      end
-    {% end %}
-
-    raise ArgumentError.new "Unknown color: #{color}"
-  end
-
-  def fore(@fore : Color)
+  # Sets the foreground color of the object to *color*.
+  def fore(color : Color)
+    @fore = color
     self
   end
 
-  def back(color : Symbol)
-    {% for name in COLORS %}
-      if color == :{{name.id}}
-        @back = ColorANSI::{{name.camelcase.id}}
-        return self
-      end
-    {% end %}
-
-    raise ArgumentError.new "Unknown color: #{color}"
-  end
-
-  def back(@back : Color)
+  # Sets the background color of the object to *color*.
+  def back(color : Color)
+    @back = color
     self
   end
 
-  def mode(mode : Symbol)
-    {% for name in MODES %}
-      if mode == :{{name.id}}
-        @mode |= MODE_{{name.upcase.id}}_FLAG
-        return self
-      end
-    {% end %}
-
-    raise ArgumentError.new "Unknown mode: #{mode}"
-  end
-
-  def on(color : Symbol)
-    back color
+  # Adds *mode* to the text's decorations.
+  def mode(mode : Mode)
+    @mode |= mode
+    self
   end
 
+  # Enables or disables colors and text decoration on this object.
   def toggle(flag)
     @enabled = !!flag
     self
   end
 
+  # Appends this object colored and with text decoration to *io*.
   def to_s(io : IO) : Nil
     surround(io) do
       io << @object
     end
   end
 
+  # Inspects this object and makes the ANSI escape codes visible.
   def inspect(io : IO) : Nil
-    surround(io) do
-      @object.inspect(io)
-    end
+    internal_io = IO::Memory.new
+    (surround(internal_io) do
```

Comment:
Why extra `()`?

Reply:
Oh, you are right. I probably forgot this when testing something.